### ISSUE TITLE
Drawio integration with gitlab

### ISSUE THREAD

**Issue by amruthapbhat** [jgraph/drawio#3098]
### Preflight Checklist

* [ ] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [ ] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Describe the bug**
Drawio integration with gitlab 

**To Reproduce**
1. Go to 'https://x.x.x.x/drawio'
2. Click on 'gitlab'

**Expected behavior**
It should redirect to gitlab url and list the repos

**Screenshots**
We are using jgraph/drawio:20.1.4 along with Gitlab 15.1 version. We are trying to integrate the both.

In gitlab we have created an application if with the below setting
<img width="710" alt="195318068-b0f3e018-9282-4d68-a76e-049617fd15a0" src="https://user-images.githubusercontent.com/24290964/195336157-f756cad1-6a7f-4e55-814e-9d58f80268f4.png">

and k8s yaml file we have given the gitlab url , gitlab secret and gitlab application id and when we try to open drawio with gitlab we get the below error.

![MicrosoftTeams-image (6)](https://user-images.githubusercontent.com/24290964/195336321-38fce3f4-4db5-4b76-84be-15d3cc4bd514.png)
 

- draw.io version [20.1.4]

**Additional context**
@davidjgraph : Can you please tell us how we can go about this issue or if there is a workaround for the same.

**Comment by amruthapbhat** [comment]
Hi can you please let me know if its an issue from gitlab or drawio side so that we can proceed accordingly

**Comment by rickywu** [comment]
I have the same problem

**Comment by Sincejunly** [comment]
I have the same problem

**Comment by m-mohamedin** [comment]
Hi,

I'm currently working on this issue. Can you please update me with details on how you setup both gitlab and draw.io (self-hosted, app setting, ...)

I've tried the integration using gitlab.com and self-hosted draw.io instance and it works.
I've used OAuth app (https://docs.gitlab.com/ee/integration/oauth_provider.html) with redirect url as https://drawio.example.com/gitlab (not it's not gitlab.html)

Note: http support will be added in the next release

**Comment by PragueJoker** [comment]
![D-Chat_20230517064239](https://github.com/jgraph/drawio/assets/16395724/881e1b59-c86f-416b-b497-275a3b42975f)

I want to know whether the api "this.redirectUri + '?getState=1'" should be implemented by ourselves. There is nowhere to say that this is to be implemented by ourselves; I didn't find any relevant instructions in the blog. I think it's just too bad up. 

If we need to implement it ourselves, what should this api return?

**Comment by m-mohamedin** [comment]
The redirectUri is implemented via GitlabAuthServlet

In the super class AbsAuth '?getState=1' is implemented, it's main purpose is to implement the state check. In non-GAE environments, the state check is disabled as it requires key-value store at the server.

TLDR, it is implemented in Java and implementation is provided along with a ready to use docker image.
